### DISCUSSION TITLE
Import servers from Active Directory / LDAP

### DISCUSSION THREAD

**Discussion by stetze** [1Remote/1Remote#431] [Ideas]
The possibility to add Computers / Servers via Active Directory / LDAP OU search string.

distinguishedName
OU=Server,DC=domainname,DC=de

![image](https://user-images.githubusercontent.com/73733823/153872564-baa9c3e9-7836-4cbc-82f4-b0373ea0ecaf.png)

The "OU" Name will be automatic added as TAG.
![image](https://user-images.githubusercontent.com/73733823/153877825-a541d7f3-cbf0-4a6a-a822-2a9087428d2a.png)

**Comment by VShawn** [comment]
Actually, It's my first time hear of this protocol.

Since I know nothing about `LDAP`, I You can make a PR if you need `LDAP`.

Just following this:
https://github.com/VShawn/PRemoteM/blob/dev/PRM.Core/Utils/mRemoteNG/mRemoteNGImporter.cs#L289-L379

**Comment by stetze** [comment]
mremoteng have That Import Function. Maybe this helps.

https://github.com/mRemoteNG/mRemoteNG/issues/211

**Comment by majkinetor** [comment]
At this moment this is IMO out of scope of this tools since LDAP/AD is enterprise thing.

**Comment by VShawn** [comment]
> At this moment this is IMO out of scope of this tools since LDAP/AD is enterprise thing.

Agree.

Even if I work on it, there is no environment for me to test it.

**Comment by majkinetor** [comment]
> Even if I work on it, there is no environment for me to test it.

I could probably create environment for this. But this is non-trivial work for more or less trivial benefit that is totally non-blocking. For example, one can search AD trivially fast with PowerShell AD cmdlets:

![image](https://user-images.githubusercontent.com/85767/154481033-dbfa685e-f481-4882-975e-8916f55f142b.png)

This could even be used with pre/post scripting currently in some capacity.

**Comment by VShawn** [comment]
:smiley: I hoping any guy can make a PR for this

**Comment by VShawn** [comment]
Can anyone give a sample ldap file for test propose?

**Comment by majkinetor** [comment]
https://github.com/nitnelave/lldap
